source/agora/network/NetworkManager.d

```diff
@@ -660,21 +662,21 @@ public class NetworkManager
     }
 
     ///
-    private bool minPeersConnected ()  pure nothrow @safe @nogc
+    private bool minPeersConnected ()  pure nothrow @safe
     {
         return this.required_peer_keys.length == 0 &&
-            this.peers.length >= this.node_config.min_listeners;
+            this.peers[].walkLength >= this.node_config.min_listeners;
     }
 
     private bool peerLimitReached ()  nothrow @safe
     {
         return this.required_peer_keys.length == 0 &&
-            this.peers.filter!(node =>
+            this.peers[].filter!(node =>
                 !this.banman.isBanned(node.address)).count >= this.node_config.max_listeners;
     }
 
     /// Returns: the list of node IPs this node is connected to
-    public NodeInfo getNetworkInfo () pure nothrow @safe @nogc
+    public NodeInfo getNetworkInfo () pure nothrow @safe
```

Comment:
This kind of modification should be another PR or commit because it's not related to this commit.